### DISCUSSION TITLE
How to add custom prettier config?

### DISCUSSION THREAD

**Discussion by andrxo** [super-linter/super-linter#2715] [Q&A] [answered]
hi team,
I wonder how i can pass my custom prettier config to super-linter?

Because i'm not satisfied with the current configuration and would like to change some settings.

Unfortunately, I wasn't able to find this setting in the documentation and source code.
Thanks.

**Comment by zkoppert** [comment, marked as answer]
Hey! Check out our doc on that here: https://github.com/super-linter/super-linter/blob/main/docs/using-rules-files.md

**Comment by coliff** [comment]
I recently ran into an issue with this too. There is no mention of specifying a custom Prettier config on the page: https://github.com/super-linter/super-linter/blob/main/docs/using-rules-files.md and no mention of it on the README.md.

I suspect it looks for a specific filename but I'm not sure what that is. (I use `.prettierrc.json` - but that appears to be ignored).

**Comment by ferrarimarco** [comment]
Can you please open an issue about this so we can investigate? thanks!

**Reply by TrentBartlem** [reply]
Done: https://github.com/super-linter/super-linter/issues/6543

**Reply by david-mcdowell-ilw** [reply]
> Can you please open an issue about this so we can investigate? thanks!

Where is .prettierrc.json supposed to be placed to be picked up by super-linter's Prettier? I have tried placing it in the root project directory or in LINTER_RULES_PATH, but both are being ignored. I also tried a .prettierrc file in the same locations. The #6543 issue that was opened is not exactly what @coliff and myself are confused about.

**Reply by ferrarimarco** [reply]
Hi, that's explained here: https://github.com/super-linter/super-linter?tab=readme-ov-file#configure-linters

**Comment by david-mcdowell-ilw** [comment]
The nature of prettier makes it difficult to tell whether your config file is being picked up and used by super-linter or not. 

I wish Prettier would produce a diff instead of just saying "Run Prettier with --write to fix"...

I was able to get a .prettierrc file working by placing it in the .github/linters/ directory (or wherever $LINTER_RULES_PATH is defined). It can be defined in YAML or JSON. Try changing the printWidth setting to validate your config file is being used.

The endOfLine setting, especially setting to "auto", which is supposed to ignore line endings, seems to be what's causing local diffs. Even when set to "auto", Prettier tries to format all files back to LF Unix-style endings, and it's causing issues with Windows CRLF code checkout.

**Reply by ferrarimarco** [reply]
> The nature of prettier makes it difficult to tell whether your config file is being picked up and used by super-linter or not.

This is actually easy to implement because you can add `--log-level debug` to the prettier invocation. Example output:

```
$ prettier --log-level debug  --check /tmp/lint/README.md
[debug] normalized argv: {"":["/tmp/lint/README.md"],"cache":false,"check":true,"color":true,"editorconfig":true,"logLevel":"debug","configPrecedence":"cli-override","debugRepeat":0,"ignorePath":[".gitignore",".prettierignore"],"plugins":[],"_":["/tmp/lint/README.md"],"__raw":{"_":["/tmp/lint/README.md"],"cache":false,"check":true,"color":true,"editorconfig":true,"log-level":"debug","config-precedence":"cli-override","debug-repeat":0,"ignore-path":[".gitignore",".prettierignore"],"plugin":[]}}
Checking formatting...
[debug] resolve config from '/tmp/lint/README.md'
[debug] loaded options `{"useTabs":false,"endOfLine":"lf","proseWrap":"always"}`
[debug] applied config-precedence (cli-override): {"proseWrap":"always","endOfLine":"lf","useTabs":false}
All matched files use Prettier code style!
```

Can you open a feature request for this? Thanks.

> I wish Prettier would produce a diff instead of just saying "Run Prettier with --write to fix"...

You're not alone :) https://github.com/prettier/prettier/issues/6885

> I was able to get a .prettierrc file working by placing it in the .github/linters/ directory (or wherever $LINTER_RULES_PATH is defined). It can be defined in YAML or JSON. Try changing the printWidth setting to validate your config file is being used.

`${LINTER_RULES_PATH}` has no effect to load Prettier configuration, as explained in https://github.com/super-linter/super-linter?tab=readme-ov-file#configure-linters

> The endOfLine setting, especially setting to "auto", which is supposed to ignore line endings, seems to be what's causing local diffs. Even when set to "auto", Prettier tries to format all files back to LF Unix-style endings, and it's causing issues with Windows CRLF code checkout.

Yeah, the `auto` setting might be tricky: https://prettier.io/docs/options#end-of-line
